github/GithubIntegration.py

```diff
@@ -212,18 +215,41 @@ def create_jwt(self, expiration: int | None = None) -> str:
         return self.auth.create_jwt(expiration)
 
     def get_access_token(
-        self, installation_id: int, permissions: dict[str, str] | None = None
+        self,
+        installation_id: int,
+        permissions: dict[str, str] | None = None,
+        repositories: list[str | int] | None = None,
     ) -> InstallationAuthorization:
         """
         :calls: `POST /app/installations/{installation_id}/access_tokens <https://docs.github.com/en/rest/apps/apps#create-an-installation-access-token-for-an-app>`
         """
+        body: dict[str, Any] = {}
+
         if permissions is None:
             permissions = {}
 
+        if repositories is not None and not isinstance(repositories, list):
+            raise GithubException(status=400, data={"message": "Invalid repositories"}, headers=None)
+
         if not isinstance(permissions, dict):
             raise GithubException(status=400, data={"message": "Invalid permissions"}, headers=None)
 
-        body = {"permissions": permissions}
+        body["permissions"] = permissions
+        if repositories:
+            repository_names = []
+            repository_ids = []
+            for r in repositories:
+                print(r, type(r))
+                if isinstance(r, str):
+                    repository_names.append(r)
+                elif isinstance(r, int):
+                    repository_ids.append(r)
+
+            if repository_names:
+                body["repositories"] = repository_names
+            if repository_ids:
+                body["repository_ids"] = repository_ids
```

Comment:
What about list comprehension?
```suggestion
            if repository_names:
                body["repositories"] = [r for r in repositories if isinstance(r, str)]
            if repository_ids:
                body["repository_ids"] = [r for r in repositories if isinstance(r, int)]
```